### DISCUSSION TITLE
Is F-Droid app maintained?

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#3043] [Q&A]
Hello,
It seems F droid app is at 1.56 still. I was wondering if do you plan to maintain it to the latest version. (Should be possible to automatically do it with each release right?)

**Comment by alextran1502** [comment]
it has been maintained by the F-Droid repo ever since. I am not sure what has been the hold up for the recent releases

**Reply by jagjordi** [reply]
should it be reported somewhere?

**Reply by jagjordi** [reply]
Anyone knows the reason why there is always a significant delay between the immich release and the f-droid release? Now its several days, usually there is 1 or 2 release difference

**Comment by kisst** [comment]
To avoid delay in releases might worth submitting to https://apt.izzysoft.de/fdroid/ , F-Droid official repo is "slow" release by design/architecture , at least compare to the release cycle of this application .

**Comment by user90210** [comment]
Still an issue

**Comment by kisst** [comment]
Well let's see, I submitted to https://gitlab.com/IzzyOnDroid/repo/-/issues/531

**Comment by kisst** [comment]
As of today F-Droid still has 1.102.3 while  1.105.1 was released  , I wonder how others handle this .
- Hold back server update too
- Hope that there isn't incompatible changes
- Download apk from GH directly
- Some other magic I am not aware of

**Reply by alextran1502** [reply]
Yeah F-Droid is not very good, I am seeing if I can move the F-Droid release to FUTO's FDroid repository so you guys can download from there since we will have more control over that

**Reply by Nordlicht-13** [reply]
Until Immich is available on FUTO, you can use Obtainium to install Immich instantly.

https://github.com/ImranR98/Obtainium

**Reply by kisst** [reply]
@alextran1502 Considering the breaking changes in the API at the latest release ,any progress here would be greatly appreciated.

**Reply by alextran1502** [reply]
@kisst we don't have control over Fdroid release cycle, we might add it to the FUTO's fdroid repository in the near future

**Reply by kisst** [reply]
To clarify I was asking about the FUTO's repo availability.

**Reply by SpiderD555** [reply]
\+ 1 for use of Futo Fdroid repo. Newpipe alredy does that and it works great. I would gladly see the same for Immich android releases.

**Reply by Magicrafter13** [reply]
I would love if there were tagged server image releases I could use to keep the server held back to the F-Droid build. I'd rather have an outdated app than one built from a non free store (personal preference), I just can't easily make my server cooperate with that.

**Reply by bo0tzz** [reply]
There are https://github.com/immich-app/immich/pkgs/container/immich-server/versions?filters%5Bversion_type%5D=tagged. If you're using the default compose installation, just set `IMMICH_VERSION` in the .env

**Reply by Magicrafter13** [reply]
> There are https://github.com/immich-app/immich/pkgs/container/immich-server/versions?filters%5Bversion_type%5D=tagged. If you're using the default compose installation, just set `IMMICH_VERSION` in the .env

I don't see any tags here which relate to the F-Droid builds?

**Reply by bo0tzz** [reply]
I'm not sure what you mean?

**Reply by Magicrafter13** [reply]
like if I could set IMMICH_VERSION to `fdroid-latest` or something so that my server version was kept in line with the F-Droid releases, due to how behind they can get sometimes. My primary usecase for Immich is Android sync but I'm not willing to use the Google Play version since it exists on F-Droid.

I can deal with it, I've done so thus far, would just be cool if there was an easy way to automatically keep those 2 mostly in sync.

**Reply by bo0tzz** [reply]
As mentioned the F-Droid builds are handled entirely by them so there's not really anywhere we could plug in a mechanism like that. That said, you shouldn't use a `latest` type tag with auto updates anyways, so I don't see it making a big difference.

Note that like mentioned downthread you could also get the app straight from the github releases with Obtainium.

**Reply by Magicrafter13** [reply]
I meant `release`, although some projects only offer `latest` so there's no choice.

I don't use Obtainium because I prefer F-Droid's build requirements, which are decently strict on keeping things libre, and different project maintainers have different app standards. Maybe I'll reconsider if I end up having like, a multi-week period of app version breakage, but so far its been manageable.

**Comment by SergioRius** [comment]
First time user. Been trying to login with the Fdroid version for the past day, thinking that I had a problem in my server. Until I noticed the difference in version with the Play app.
It's ironical coming here trying to de-google and being tied to their app store. HeHeHeHe...
Would be a good idea to base on an independent store.

**Reply by alextran1502** [reply]
@SergioRius if you want to help the most people, it has to be the most approachable possible for distribution methods 😉

**Reply by SergioRius** [reply]
Totally agree. And BTW, thank you very much for your work.
I hope there's no problem with joojle on their store.

**Comment by Nordlicht-13** [comment]
Get it directly from Github via Obtainium:
https://github.com/ImranR98/Obtainium

**Comment by kisst** [comment]
Just noticed that the application is now published to Google Play Store too, and there the latest version is available, 1.115.0 while F-droid is still stuck on 1.112.1,  I think FUTO is the way to go, who can publish it to https://app.futo.org/fdroid/repo/ ?

**Comment by EinToni** [comment]
@alextran1502 Is there any update regarding adding the app to alternative fdroid repos (izzi/futo)? 
The current fdroid app is again over a week behind :/

**Reply by DVSProductions** [reply]
Now it's already a whole month. How is this possible?

**Reply by NicholasFlamy** [reply]
F-Droid is responsible for their release schedule. I recommend Obtainium (which can also work with F-Droid apps).
I'm working on a PR for something that would help sync Obtainium with the installed server version of immich.

**Reply by SISheogorath** [reply]
Please don't expect wonders: https://gitlab.com/fdroid/wiki/-/wikis/FAQ#how-long-does-it-take-for-my-app-to-show-up-on-website-and-client

Basically unless someone manually does something, the earliest release of an app is within 72 hours. It also depends if the app actually makes it into a release cycle, and if there is enough build time for all apps. If other apps take longer to build, it's possible that a build is pushed back in another release cycle. (See: https://gitlab.com/fdroid/fdroiddata/-/issues/3547#note_2497395580)

In other words: Just be patient :) 1.33.0 is released by now.

**Reply by kisst** [reply]
@NicholasFlamy  @SISheogorath  I think you both are missing the point, the request isn't about push quicker to the F-droid official repo, it's well understood that isn't possible, the ask is to publish into FUTO's own F-droid repo ( https://app.futo.org/fdroid/repo/ )  since now Immich is part of FUTO for more then a year by now ( https://immich.app/blog/2024/immich-core-team-goes-fulltime/ ) it would be the logical thing to do.

**Reply by NicholasFlamy** [reply]
Yeah, I guess that's a good point. I will point out that their F-Droid repo is always days behind, and I've stopped using it for GrayJay and FUTO Keyboard because of that.
